### PR TITLE
refactor tests

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #43743 Update aliasing in tensorexpr fuser
* #43742 use types in the IR instead of vmap
* #43636 Add inline call to profiling executor pipeline
* #43635 Use prim::TensorExprGroup interned symbol
* #43634 Update requires grad property
* #43633 Specialize optionals for grad_sum_to_size
* #43632 Add undefined specializations in backward
* **#43631 refactor tests**
* #43630 Refactor pass to class
* #43629 Add API for unexecuted op

I added a new test for just profiler stuff - I don't think the test should go in test_jit.py. Maybe this should just go in test_tensorexpr_fuser, but I'm not really testing tensorexpr stuff either... LMK

Differential Revision: [D23358810](https://our.internmc.facebook.com/intern/diff/D23358810)

### CODE REVIEW

File: test/jit/test_profiler.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+pytorch_test_dir = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+sys.path.append(pytorch_test_dir)
+from torch.testing._internal.jit_utils import JitTestCase
+
+if __name__ == '__main__':
+    raise RuntimeError("This test file is not meant to be run directly, use:\n\n"
+                       "\tpython test/test_jit.py TESTNAME\n\n"
+                       "instead.")
+
+class TestProfiler(JitTestCase):
```
Comment (ZolotukhinM) [inline test/jit/test_profiler.py:16]: Is it used anywhere?

**Review by ZolotukhinM — APPROVED** [review]
Apart from the previous comment, the change looks good!

File: test/jit/test_profiler.py (hunk 1 above)
Comment (eellison) [reply test/jit/test_profiler.py:16]: it's imported in test_jit.py, test cases are purposefully empty

Comment (ZolotukhinM) [reply test/jit/test_profiler.py:16]: What's the purpose of empty tests?

Comment (eellison) [reply test/jit/test_profiler.py:16]: so the next PR that adds tests here has a smaller set of changes

**Pull request closed without merging**